src/app/core/services/albums.service.ts

```diff
@@ -0,0 +1,88 @@
+import { DestroyRef, Injectable } from '@angular/core';
+import {
+  AngularFirestore,
+  AngularFirestoreCollection,
+} from '@angular/fire/compat/firestore';
+import { UsersService } from './users.service';
+import { AuthenticationService } from './authentication.service';
+import { firstValueFrom, map, throwError } from 'rxjs';
+import { Album, AlbumWithId, UserAlbum } from '../models/album.interface';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AlbumsService {
+  private albumsPath = '/albums';
+  albumsRef: AngularFirestoreCollection<Album>;
+  constructor(
+    private usersService: UsersService,
+    private authenticationService: AuthenticationService,
+    private destroyRef: DestroyRef,
+    private db: AngularFirestore,
+  ) {
+    this.albumsRef = db.collection(this.albumsPath);
+  }
+
+  getAll() {
+    return this.albumsRef.snapshotChanges().pipe(
+      takeUntilDestroyed(this.destroyRef),
+      map((albums) => {
+        return albums.map(
+          (game): AlbumWithId => ({
+            id: game.payload.doc.id,
+            ...game.payload.doc.data(),
+          }),
+        );
+      }),
+    );
+  }
+
+  get(id: string) {
+    return this.albumsRef.doc(id).valueChanges();
+  }
+
+  async addUser(albumId: string) {
+    const album = await firstValueFrom(this.get(albumId));
+    const authUser = await firstValueFrom(this.authenticationService.getUser());
+
+    if (!album || !authUser?.uid) {
+      throwError(() => new Error('Could not find album or user'));
+      return;
+    }
+
+    const data = album;
+    data.saved_by.push(authUser.uid);
+    await this.albumsRef.doc(albumId).update(data);
```

Comment:
Hmm. Why do you resolve that? you don't do anything with the value and promises are running without  resolving them. So this should work ?

Also you await when you use `addUser` (I think I saw that somwhere) which would return nothing. So you would remove the await there too.

And seeing you using await/asycn so extensively is great. Let's the the theory straight now:

You absolutely have to learn about the js event loop if you didn't do that yet. My two favourites:

+ https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/
+ https://www.youtube.com/watch?v=8aGhZQkoFbQ